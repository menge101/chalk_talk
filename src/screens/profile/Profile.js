import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';
import PropTypes from 'prop-types'


export class ProfileScreen extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Image
          source={this.props.pic}
          style={{width: 100, height: 100}}
        />
        <Text>
          {this.props.hashName}
        </Text>
        <Text>
          {this.props.motherKennel}
        </Text>
      </View>
    );
  }
}

ProfileScreen.propTypes = {
  pic: PropTypes.object,
  hashName: PropTypes.string.isRequired,
  motherKennel: PropTypes.string.isRequired
}

ProfileScreen.defaultProps = {
    pic: null
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});