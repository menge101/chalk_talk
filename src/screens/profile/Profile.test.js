import React from 'react';
import renderer from 'react-test-renderer';

import { ProfileScreen } from './Profile.js'

describe('<ProfileScreen />', () => {
  it('matches rendering snapshot', () => {
    let pic = require('./pic.jpg')
    let hashName = 'Kunt-Fu Weasel'
    let motherKennel = 'Pittsburgh Hash House Harriers'
    const tree = renderer.create(
      <ProfileScreen
        pic={pic}
        hashName={hashName}
        motherKennel={motherKennel}
      />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});