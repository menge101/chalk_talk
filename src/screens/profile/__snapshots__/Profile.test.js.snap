// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileScreen /> matches rendering snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text>
    Kunt-Fu Weasel
  </Text>
  <Text>
    Pittsburgh Hash House Harriers
  </Text>
</View>
`;

exports[`<ProfileScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text>
    Kunt-Fu Weasel
  </Text>
  <Text>
    Pittsburgh Hash House Harriers
  </Text>
</View>
`;
